i18n-helpers/src/lib.rs

```diff
@@ -333,17 +360,105 @@ fn is_comment_skip_directive(html: &str) -> bool {
+                    let mut stack = ScopeStack::new();
+                    if let Ok(ops) = ps.parse_line(text, &ss) {
+                        // Temporary buffer to concatenate sequential comments
+                        let mut translate = vec![];
+
+                        for (range, scope) in ScopeRangeIterator::new(&ops, text) {
+                            // Calculate line number of the range
+                            let range_line = if range.start == 0 {
+                                *text_line
+                            } else {
+                                text_line + text[0..range.start].lines().count() - 1
+                            };
+
+                            let _ = stack.apply_with_hook(scope, |op, scopes| {
+                                let range = range.clone();
+                                if !range.is_empty() {
+                                    let is_translate = match op {
+                                        BasicScopeStackOp::Push(x) => {
+                                            is_translate_scope(x)
+                                                | scopes.iter().any(|x| is_translate_scope(*x))
+                                        }
+                                        BasicScopeStackOp::Pop => {
+                                            scopes.iter().any(|x| is_translate_scope(*x))
+                                        }
+                                    };
+
+                                    let text = &text[range];
+
+                                    if is_translate {
+                                        translate.push((
+                                            range_line,
+                                            Event::Text(CowStr::Borrowed(text)),
+                                        ));
+                                    } else {
+                                        if !translate.is_empty() {
+                                            ret.push(Group::TranslateOwned(std::mem::take(
+                                                &mut translate,
+                                            )));
+                                        }
+                                        ret.push(Group::SkipOwned(vec![(
+                                            range_line,
+                                            Event::Text(CowStr::Borrowed(text)),
+                                        )]));
+                                    }
+                                }
+                            });
+                        }
+                        if !translate.is_empty() {
+                            ret.push(Group::TranslateOwned(std::mem::take(&mut translate)));
+                        }
+                    } else {
+                        // If parse is failed, the whole codeblock should be translated.
+                        ret.push(Group::Translate(&events[idx..idx + 1]));
```

Comment:
This could probably also be moved to the top of the loop — perhaps using a `let-else`: https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html.